espnet2/enh/decoder/codec_decoder.py

```diff
@@ -0,0 +1,89 @@
+import torch
+import torch.nn.functional as F
+import torchaudio.transforms as T
+
+from espnet2.enh.decoder.abs_decoder import AbsDecoder
+from espnet2.speechlm.tokenizer.codec_tokenizer import CodecTokenizer
+
+
+class CodecDecoder(AbsDecoder):
+    """Codec decoder for speech enhancement and separation"""
+
+    def __init__(
+        self,
+        channel: int,  # this is the expected output size of the continuous codec output
+        codec_choice: str,
+        codec_fs: int,
+        sample_fs: int,
+        device: str = "cpu",
+        dump_audio: bool = False,
+        checkpoint_path: str = None,
+        config_path: str = None,
+        max_token_per_frame: int = 32,
+    ):
+        super().__init__()
+        self.codec = CodecTokenizer(
+            codec_choice=codec_choice,
+            codec_fs=codec_fs,
+            device=device,
+            dump_audio=dump_audio,
+            checkpoint_path=checkpoint_path,
+            config_path=config_path,
+        )
+        self.codec_fs = codec_fs
+        self.sample_fs = sample_fs
+        if self.codec_fs != self.sample_fs:
+            self.dac_sampler = T.Resample(codec_fs, sample_fs).to(device)
+
+        for param in self.codec.parameters():
+            param.requires_grad = False
+        self._output_dim = channel
+
+    def forward(self, input: torch.Tensor, ilens: torch.Tensor, fs: int = None):
+        """Forward.
+
+        Args:
+            input (torch.Tensor): spectrum [Batch, T, F]
```

Comment:
It is better to consider a future extension to support multi-channel input speech at this stage. Currently the interface always assume the input to be single-channel.